File: Haskell/eyeofsauron.hs
import Data.List.Split (splitOn)

count :: String -> Int
count [] = 0
count ('|':xs) = 1 + count xs
count (_:xs) = count xs

check :: Int -> Int -> String
check x y
    | x == y = "correct"
    | otherwise = "fix"

main :: IO ()
main = do
    input <- getLine
    let [left, right] = splitOn "()" input
    let x = count left
    let y = count right
        ans = check x y
    putStrLn ans

File: python/eyeofsauron.py
x = input()

partitioned = x.partition("()")
countLeft = 0
countRight = 0

for left in partitioned[0]:
    countLeft+=1
    
for right in partitioned[2]:
    countRight+=1

if countRight == countLeft:
    print("correct")
else:
    print("fix")
